pkg/reconciler/v1alpha1/trigger/trigger.go

```diff
@@ -360,11 +358,11 @@ func (r *reconciler) getChannel(ctx context.Context, b *v1alpha1.Broker, ls labe
 
 // getService returns the K8s service for trigger 't' if exists,
 // otherwise it returns an error.
-func (r *reconciler) getService(ctx context.Context, t *v1alpha1.Trigger) (*corev1.Service, error) {
+func (r *reconciler) getBrokerFilterService(ctx context.Context, b *v1alpha1.Broker) (*corev1.Service, error) {
 	list := &corev1.ServiceList{}
 	opts := &runtimeclient.ListOptions{
-		Namespace:     t.Namespace,
-		LabelSelector: labels.SelectorFromSet(resources.ServiceLabels(t)),
+		Namespace:     b.Namespace,
+		LabelSelector: labels.SelectorFromSet(brokerresources.FilterLabels(b)),
 		// Set Raw because if we need to get more than one page, then we will put the continue token
 		// into opts.Raw.Continue.
 		Raw: &metav1.ListOptions{},
```

Comment:
I don't think this works with the controller runtime client got in-memory get. Looking into its code I couldn't find any usage of this listOption. could you check once?

Reply:
I want to leave this in so that we match the contract, even if Controller Runtime's client doesn't use it in the current implementation, I think the contract says it could.